src/Missings.jl

```diff
@@ -165,4 +165,30 @@ function levels(x)
     levs
 end
 
+struct PassMissing{F} <: Function end
+
+@generated (::PassMissing{F})(xs...;kw...) where {F} =
+    :(any(ismissing, xs) || any(ismissing, values(values(kw))) ? missing : F(xs...; kw...))
```

Comment:
Hmmm, the inclusion of keyword arguments here seems off to me; what if `missing` is a totally valid thing to pass as a keyword argument, but now it's interacting weird w/ `PassMissing` because it always makes my result `missing`?

Reply:
I think my stab at this definition would be more along the lines of:
```julia
struct PassMissing{F} <: Function                                                                                             
    f::F                                                                                                                      
end                                                                                                                           
                                                                                                                              
function (f::PassMissing{F})(x) where {F}                                                                                     
    if @generated                                                                                                             
        return x === Missing ? missing : :(f.f(x))                                                                            
    else                                                                                                                      
        return x === missing ? missing : f.f(x)                                                                               
    end                                                                                                                       
end                                                                                                                           
                                                                                                                              
function (f::PassMissing{F})(xs...; kw...) where {F}                                                                          
    if @generated                                                                                                             
        for T in xs                                                                                                           
            T === Missing && return missing                                                                                   
        end                                                                                                                   
        return :(f.f(xs...; kw...))                                                                                           
    else                                                                                                                      
        return any(ismissing, xs) ? missing : f.f(xs...; kw...)                                                               
    end                                                                                                                       
end 
```
with specialize methods for 1 argument, 2 argument, maybe 3.

Reply:
Interesting. Though with generated functions you don't need to specialize on the number of arguments.

Also why do you think keyword arguments should be excluded from the missingness check?

Reply:
The `if @generated` part is just a future-proof against a time when you're able to statically compile a julia program into an executable that can run without LLVM (i.e. w/o runtime compilation). The `else` block just provides a path that would be executed if the compiler encountered an already non-generated/compiled method for the given arguments at runtime.

For keyword arguments, it just feels more arbitrary. Like, it's hard to imagine a case where I'd be `map`ing over some `Vector{Union{Float64, Missing}}` and `Vector{Union{Int, Missing}}` and be doing something like `map((x, y)->round(x; digits=y), zip(A, B))`. i.e. when would I maybe pass a value vs. maybe pass a `missing` as a keyword argument? Keyword arguments also tend to use various "sentinel" values as signals or special values for the function to use, including `missing`; the danger there being that someone writes a function `foo(x...; sentinel=nothing)`, but when the user tries to call `Missings.propagate(foo)(x; sentinel=missing)`, the entire result comes back `missing` instead of passing `missing` on to `foo` as a valid sentinel value.

If anything, I'd say we leave keyword arguments out of the mix for now since they can always be added later.

My main issue with `passmissing` is that `pass` usually has a connotation of "ignore" or "do nothing" (see python's [pass](https://docs.python.org/2.0/ref/pass.html)), whereas we're not really ignoring `missing` values, we're propagating them.

Reply:
> For keyword arguments, it just feels more arbitrary. Like, it's hard to imagine a case where I'd be `map`ing over some `Vector{Union{Float64, Missing}}` and `Vector{Union{Int, Missing}}` and be doing something like `map((x, y)->round(x; digits=y), zip(A, B))`. i.e. when would I maybe pass a value vs. maybe pass a `missing` as a keyword argument? Keyword arguments also tend to use various "sentinel" values as signals or special values for the function to use, including `missing`; the danger there being that someone writes a function `foo(x...; sentinel=nothing)`, but when the user tries to call `Missings.propagate(foo)(x; sentinel=missing)`, the entire result comes back `missing` instead of passing `missing` on to `foo` as a valid sentinel value.

I don't think there are any examples of functions using `missing` as a sentinel currently, right? We rather want people to use `nothing` for that.

I see your point about keyword arguments generally being about options, while the data is passed as positional arguments, but I'd find it problematic to make this a rule. Overall it doesn't seem we have strong use cases for either behavior, and in such situations I tend to favor the simplest rule (i.e. "return `missing` if one of the arguments is `missing`").

> If anything, I'd say we leave keyword arguments out of the mix for now since they can always be added later.

The problem is, changing that would be breaking. The only way to be able to choose any behavior later is to throw an error if a keyword argument is `missing`.

Reply:
> The only way to be able to choose any behavior later is to throw an error if a keyword argument is missing.

We could define lift for functions that do not take keyword arguments for the time being. It is easy enough to wrap a function requiring kwargs in an anonymous function.